docs/DeveloperGuide.md

```diff
@@ -1,38 +1,235 @@
 # Developer Guide
 
-## Acknowledgements
+---
+### Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+1. Command, Parser and UI java files are adapted from one of our group member's [Duke Project](https://github.com/MingEn82/tp)
+---
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+---
+### Architecture
 
+![](../out/docs/uml/Architecture/architecture.svg)
 
-## Product scope
-### Target user profile
+**Main Components of LifeTracker**
+
+`LifeTracker` is where users will access the application from. It is responsible for:
+
+- At app launch: Initialise the Databases and UI classes
+- At app closure: Saves all updated data to databases
+
+The rest of the Application consists of three components:
+
+- `UI`: Reponsible for I/O between user and application
+- `Command`: Handles execution of user inputs
+- `Storage`: Stores information of the user and meals eaten
+
+---
+### UI Component
+The UI is made up of three classes, `GeneralUi`, `CalorieUi` and `WeightUi`, and together,
+they improve on the accessibility of the application.
+
+The `UI` component,
+- Displays what are the possible inputs at any one time
+- Allows the user to check that a value added is correct
+- Informs the user whenever there is an invalid input
+
+### Storage Component
+
+![storage-class-diagram](./uml/storage-class-diagram.png)
```

Comment:
![image](https://user-images.githubusercontent.com/37783236/229006481-05b91fe1-ac8a-44ae-8c4e-139d0083ba68.png)

Storage-class-diagram is not showing up. If you are using plantUML, simply copy paste the image over to your DG from the UML file.